File: returnLongestNonRepeatSubString/python/hello.py
def returnLongestNonRepeatSubString(s):
    if not s:
        return ""
    subCollection = []
    currentSub = ""
    for c in s:
        if (currentSub.find(c) == -1):
            currentSub += c
        else:
            subCollection.append(currentSub)
            currentSub = c
    
    if currentSub:
        subCollection.append(currentSub)
    
    longest = ""
    for sub in subCollection:
        if (len(sub) > len(longest)):
            longest = sub
            
    return longest

File: returnLongestNonRepeatSubString/c++/hello.cpp
string
returnLongestNonRepeatSubString (const string & s)
{
    if (s.empty ())
    {
        return "";
    }
    std::vector <string> subCollection;
    string currentSub;
    for (int i = 0; i < s.length (); i++)
    {
        char c = s[i];
        if (currentSub.find (c) == -1)
        {
            currentSub += c;
        }
        else
        {
            subCollection.push_back (currentSub);
            currentSub = string (1, c);
        }
    }

    if(!currentSub.empty())
    {
        subCollection.push_back (currentSub);
    }

    string longest = "";

    for (int i = 0; i < subCollection.size(); i++)
    {
        string sub = subCollection.at(i);
        if (sub.length() > longest.length())
        {
            longest = sub;
        }
    }
    return longest;
}
